scripts/deps.sh

```diff
@@ -17,18 +17,57 @@
 #See the License for the specific language governing permissions and
 #limitations under the License.
 
-# This script runs the correct godep sequences for snap and built-in plugins
-# This will rebase back to the committed version. It should be run from snap/.
-ctrl_c()
-{
-  exit $?
+set -e
+set -u
+set -o pipefail
+
+__dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+__proj_dir="$(dirname "$__dir")"
+
+# shellcheck source=scripts/common.sh
+. "${__dir}/common.sh"
+
+detect_go_dep() {
+  [[ -f "${__proj_dir}/Godeps/Godeps.json" ]] && _dep='godep'
+  [[ -f "${__proj_dir}/glide.yaml" ]] && _dep='glide'
+  [[ -f "${__proj_dir}/vendor/vendor.json" ]] && _dep='govendor'
+  _info "golang dependency tool: ${_dep}"
+  echo "${_dep}"
 }
-trap ctrl_c SIGINT
 
-# Godep
-echo "Getting godep if not found"
-go get github.com/tools/godep
+install_go_dep() {
+  local _dep=${_dep:=$(_detect_dep)}
+  _info "ensuring ${_dep} is available"
+  case $_dep in
+    godep)
+      _go_get github.com/tools/godep
+      ;;
+    glide)
+      _go_get github.com/Masterminds/glide
+      ;;
+    govendor)
```

Comment:
Only other comment I have would be to remove glide and govendor from install_go_dep and restore_go_dep and reintroduce glide with the other PR.